reflection.md

```diff
@@ -2,17 +2,59 @@
 
 ## Which operators were easiest to implement and why?
 
+Relatively, the `||`, `~`, and `*` operators were the easiest to implement. They
+were all just calling one function, and the hardest part was figuring out how to
+call them and in what format.
+
 ## Which operators were most difficult to implement and why?
 
+Probably the implicit conversions and the `<+>` operator. The implicit
+conversions because I did not understand their purposes at first, and that
+prevented me from implementing it correctly.
+
+For chars, it was relatively easy to figure out that a simple `Literal` call
+seemed to suffice. However, it took a little more regarding the strings because
+I was experimenting different ways of getting the string to become chars that I
+could then use the `Literal` or `charToRegex` on. I tried converting the string
+into a list or sequence (`toList` and `toSeq`), but I finally settled on using
+a `foldLeft` directly on the string. After trial and error, I figured out the
+accumulator/base case should be an `EPSILON`, and that was that.
+
+For the `<+>` operator, what threw me off was what to do after the first `~`,
+and solified this operators difference from the Kleene Star operator.
+I had to revisit stuff from 81 regarding how repetitions were defined so that
+I could model what the operator should do. Once I figured that out, the operator
+was pretty easy to code.
+
 ## Comment on the design of this internal DSL
 
 Write a few brief paragraphs that discuss:
    + What works about this design? (For example, what things seem easy and
    natural to say, using the DSL?)
+
+   It's easy to do the regex as defined in our syntax. So
+   `"42" || ( ('a' <*>) ~ ('b' <+>) ~ ('c'{3}))` is easy to say and match.
+
    + What doesn't work about this design? (For example, what things seem
    cumbersome to say?)
+
+   Some standard regex is hard to do. If I just wanted patterns that were made
+   up of the 26 lowercase English characters, instead of writing `[a-z]*`, in
```

Comment:
I agree; this is something I do all the time when I write regular expressions.

I also miss Python's verbose mode

```python
re.compile("""
    ^    # Start of the string
    [a-f0-9]    # hex character
    (\.|[a-f0-9])* #series of periods/hex characters
    8888$    # end with four 8s, and nothing else
    """, re.VERBOSE)
```

while this isn't super useful in that contrived example, but for complex regular expressions it is a huge boon. The other thing that would be really nice is regex interpolation, which you can kind of get with Python 3.6

```
hexNumber = r"[a-fA-F0-9]"

re.compile(rf"^{hexNumber}(\.|{hexNumber})*8888$")
```

using string interpolation.